### PR TITLE
docs: fix PHPDoc and comments in Models

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- replace  type mixed. See #6310
- fix comments

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by paulbalandan — APPROVED** [review]

File: system/BaseModel.php [hunk 1]
```diff
@@ -995,9 +996,9 @@ public function delete($id = null, bool $purge = false)
 
     /**
      * Permanently deletes all rows that have been marked as deleted
-     * through soft deletes (deleted = 1)
+     * through soft deletes (deleted = 1).
      *
-     * @return mixed
+     * @return bool|string Returns a string if in test mode.
```
Comment (MGatner) [inline system/BaseModel.php:1001]: These are annoying to support. I would like to see in v5 some kind of Database Proxy that passes requests through to the driver for most cases but "captures" the actual SQL query for Test Mode. I think that would be a much better approach than trying to adjust all our return types.

File: system/BaseModel.php [hunk 2]
```diff
@@ -1170,11 +1172,11 @@ protected function doProtectFields(array $data): array
...
-     * Sets the date or current date if null value is passed
+     * Sets the date or current date if null value is passed.
      *
      * @param int|null $userData An optional PHP timestamp to be converted.
      *
-     * @return mixed
+     * @return int|string
      *
      * @throws ModelException
      */
```
Comment (MGatner) [inline system/BaseModel.php:1182]: 👇#6815

File: system/BaseModel.php [hunk 3]
```diff
@@ -1397,7 +1399,7 @@ public function getValidationRules(array $options = []): array
     }
 
     /**
-     * Returns the model's define validation messages so they
+     * Returns the model's define validation messages, so they
```
Comment (MGatner) [inline system/BaseModel.php:1402]:
```suggestion
     * Returns the model's validation messages, so they
```

**Review by MGatner — CHANGES_REQUESTED** [review]

File: system/BaseModel.php (hunk 1 above)
Comment (kenjis) [reply system/BaseModel.php:1001]: Yes, the return types are not good, should be `bool` only.

File: system/BaseModel.php (hunk 2 above)
Comment (kenjis) [reply system/BaseModel.php:1182]: I sent a PR #6825

**Comment by kenjis** [comment]
What's these errors?

```
There were 479 errors:

1) CodeIgniter\Commands\CreateDatabaseTest::testCreateDatabase
CodeIgniter\Database\Exceptions\DatabaseException: Unable to connect to the database.
Main connection [SQLSRV]: Call to undefined function CodeIgniter\Database\SQLSRV\sqlsrv_configure()

/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/BaseConnection.php:418
/home/runner/work/CodeIgniter4/CodeIgniter4/system/Database/Database.php:79
/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/Commands/CreateDatabaseTest.php:48
```
https://github.com/codeigniter4/CodeIgniter4/actions/runs/3404234441/jobs/5664964160

**Comment by kenjis** [comment]
@paulbalandan Do you know anything?

```
Run shivammathur/setup-php@v2
/usr/bin/bash /home/runner/work/_actions/shivammathur/setup-php/v2/src/scripts/run.sh

==> Setup PHP
✓ PHP Installed PHP 8.2.0-dev (94f6a0c1dc8ab939b37e93b448bc2edce8c48206)

==> Setup Extensions
✓ imagick Enabled
✗ sqlsrv Could not install sqlsrv on PHP 8.2.0-dev
✓ gd Enabled
✓ sqlite3 Enabled
✓ redis Enabled
✓ memcached Enabled
✓ oci8 Installed and enabled
oci8 Click to read the oci8 related license information 
✓ pgsql Enabled
```

**Comment by paulbalandan** [comment]
Not sure, but can you try changing the extension to `sqlsrv-5.10.1`?

**Comment by kenjis** [comment]
@paulbalandan Passed!

**Pull request merged**